### PR TITLE
[CDAP-15915] integrating validation with dynamic plugin filters

### PR DESCRIPTION
**Pull request by itsanudeep**
Plugin properties that are hidden to the user will not be part of validation call.

### CODE REVIEW

File: cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts [hunk 1]
```diff
@@ -257,3 +258,44 @@ export function filterByCondition(
...
+    processedConfigurationGroup.configurationGroups,
+    widgetJson,
+    pluginProperties,
+    values
+  );
+  // figuring out which properties are hidden
+  filteredConfigurationGroups.forEach((group: IFilteredConfigurationGroup) => {
+    if (group.show) {
+      group.properties.forEach((property) => {
+        if (property.show) {
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts:292]: Possibly silly question, but if it is hidden shouldn't it be !property.show?

File: cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js [hunk 2]
```diff
@@ -537,11 +537,14 @@ class HydratorPlusPlusNodeConfigCtrl {
         }
       }, (err) => {
         vm.validating = false;
+        const callFailureMsg = 'Internal error occured while validating.';
```
Comment (yukiej) [inline cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js]: spelling nit: should be "occurred

File: cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts [hunk 3]
```diff
@@ -257,3 +258,44 @@ export function filterByCondition(
...
+          hiddenProps.add(name);
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts]: should this be property.name?

**Review by yukiej — APPROVED** [review]
lgtm!

File: cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts [hunk 4]
```diff
@@ -257,3 +258,45 @@ export function filterByCondition(
...
+    'configuration-groups'
+  );
+  const widgetOutputs: IWidgetProperty[] = objectQuery(widgetJson, 'outputs');
+  const values: IPropertyValues = objectQuery(nodeInfo, 'plugin', 'properties');
+  const processedConfigurationGroup: IProcessedConfigurationGroups = processConfigurationGroups(
+    pluginProperties,
+    widgetConfigurationGroup,
+    widgetOutputs
+  );
+  const filteredConfigurationGroups: IFilteredConfigurationGroup[] = this.filterByCondition(
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts]: what is `this` refer to? a function will create its own `this` reference. `filterByCondition` is defined in the same file, so you should be able to call it by the function name

File: cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts [hunk 5]
```diff
@@ -30,6 +30,7 @@ import {
 import flatten from 'lodash/flatten';
 import difference from 'lodash/difference';
 import { objectQuery, isMacro } from 'services/helpers';
+import { IProcessedConfigurationGroups, processConfigurationGroups } from './index';
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts]: import path should use components

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by itsanudeep** [comment]
Build and Test : https://builds.cask.co/browse/CDAP-UDUT414-8
RAT and Checkstyle: https://builds.cask.co/browse/CDAP-DRC9527-9

**Pull request merged**